Replay commits

ssk2 opened the pull request:
Bringing this repo up to speed with the private fork.

colin-msphere committed:
Add Support for Plugins not in Update Center

Add a way to include plugins that are not available through the Jenkins
Update Center. This is done by providing an optional URL portion to the
entries inside the plugins array. If the URL is present, the Update
Center is skipped and the file downloaded directly from the URL.

Commit:
(fixup) reformat github-api line

Commit:
include the marathon plugin in the readme

colin-msphere committed:
Add Helper Methods for XML Operations

Add helper methods:
1. get the root of an XML document
2. find and replace the text of an XML element

colin-msphere committed:
Add Optional JENKINS_INSTANCE_URL Environment Var

Allow the Jenkins instance URL to be passed in, otherwise populated it
the old way with marathon host and port.

colin-msphere committed:
Migrate Others to Helpers

Migrate the remaining methods to use the XML helpers.

colin-msphere committed:
Change Argument Name

Change the argument name from "text" to "new_text". This is to avoid
confusion with the Element attribute "text" within the same method.

colin-msphere committed:
Use os.getenv

Use os.getenv(key) over the other way. This defaults to returning
None when the key is not found.

colin-msphere committed:
Update Plugins to Latest Versions

This upgrades almost all plugins to the most recent versions.
This did not upgrade scm-api, as it was recently bumped from
0.2 to 1.1 and may have issues.

colin-msphere committed:
Disable DNS Multicast

This fixes a typo that prevented DNS multicast from being disabled.

colin-msphere committed:
Rename JENKINS_INSTANCE_URL to JENKINS_ROOT_URL

Change the name of the optional environment variable
JENKINS_INSTANCE_URL to a name that is less ambiguous.

Commit:
add artifactory plugin

colin-msphere committed:
Rename Local Variable and Set Default

Change the local variable from jenkins_instance_url to jenkins_root_url
to avoid confusion and be consistent with the environment variable.

Leverage the "default" parameter of os.getenv to set the root URL to
the Marathon host and Nginx port if the environment variable is not
set. This avoids checking for None later.

Commit:
add role-strategy plugin

This commit adds the Role Strategy plugin, which provides a role-based
strategy for user management.

Commit:
add saml plugin for onelogin integration

This commit adds the SAML plugin to Velocity so that we can integrate
with OneLogin.

Commit:
add Embeddable Build Status plugin

Adds the ability to add build status badges to a project.

Commit:
add cloudbees folders plugin

Commit:
add github pull request builder plugin

ssk2 committed:
Bump to 1.651.1 LTS version

Commit:
add timestamper plugin

Commit:
add build timeout plugin

Commit:
add mailer plugin

Commit:
add build-name-setter plugin

Commit:
s/jenkins-ci.org/jenkins.io/

colin-msphere committed:
Add Ability to Package Plugins into WAR

This commit adds the ability to create a single WAR file with all
plugins bundled inside the WAR. This helps with plugin upgrades
and adding new plugins after the initial deployment.

colin-msphere committed:
Remove Version from Marathon HPI

Remove the version suffix of the marathon.hpi to match all other
plugins.

colin-msphere committed:
Add Constructed WAR File

Use the file created by maven instead of the hosted Jenkins
distribution.

colin-msphere committed:
Consolidate ENV Vars

Remove unnecessary environment variables.

colin-msphere committed:
Remove install_plugins

Delete the script to install plugins.

colin-msphere committed:
Use Jenkins LTS Version

Fix Jenkins version to use an LTS version.

colin-msphere committed:
Fix Repo URL

Fix Jenkins repo URL. The old one performs a redirect to the
one in this commit.

colin-msphere committed:
Upgrade Jenkins to latest LTS

Upgrade to 1.651.2 due to security issues.

colin-msphere committed:
Update README

I forgot the README.

colin-msphere committed:
Update Marathon Deployment plugin

Upgrade marathon plugin from 1.2 to 1.2.1

colin-msphere committed:
Update README

Commit:
add release script to version control

Prior to this commit, the build script for this repository was stored in
Jenkins. This moves it to the `scripts/` directory so that it can simply
be called in Jenkins, and adds it to the .dockerignore file so that it
won't be part of the final Docker image.

colin-msphere committed:
Upgrade DIND Image

Bump DIND image from 0.2.x to 0.3.0.

colin-msphere committed:
Allow Undefined Parameters

At this time, allow undefined parameters to be injected into jobs. This
restores the functionality of the GitHub Pull Request Builder and other
plugins that inject new, undefined build parameters into jobs.

colin-msphere committed:
Support SECURITY-170

This commit upgrades plugins to support SECURITY-170 and
no longer allow build parameter injection.

colin-msphere committed:
Add matrix-project

The matrix-project version that was released with 1.651.2 is
not compatible with SECURITY-170. This commit adds matrix-project
plugin as a managed plugin and uses the latest version that is
compatible.

ssk2 committed:
Update dind-image version

ssk2 committed:
Update changelog.

ssk2 committed:
s/teamcity/jenkins/

rji commented on pom.xml:
```diff
@@ -0,0 +1,385 @@
+
+	<groupId>com.mesosphere.velocity</groupId>
+	<artifactId>velocity</artifactId>
+	<version>1.2-SNAPSHOT</version>
```
Assuming that this is going to be the 1.0.0 release, we should update the version here

rji commented on pom.xml:
```diff
@@ -0,0 +1,385 @@
+		<dependency>
+			<groupId>org.jenkins-ci.main</groupId>
+			<artifactId>jenkins-war</artifactId>
+			<version>1.658</version>
```
This should be the latest non-2.x LTS release

ssk2 replied:
Why is this 1.658 out of interest? Looks like the most recent stable was 1.651.3.

rji commented on pom.xml:
```diff
@@ -0,0 +1,385 @@
+							<goal>wget</goal>
+						</goals>
+						<configuration>
+							<url>https://s3.amazonaws.com/downloads.mesosphere.io/velocity/jenkins/plugins/marathon-1.2.0.hpi</url>
```
This is available in the Jenkins update center now

rji commented:
A couple of inline comments, plus we shouldn't need `scripts/plugin_install.sh` since we moved to Maven for plugin management.

rji commented on CHANGELOG.md:
```diff
@@ -1,4 +1,7 @@
 # Changes
+## v1.0.0 (2016-03-24)
```
We should probably make the v1.0.0 release date today's date (2016-07-07)

ssk2 replied:
Done (will force push up in a minute).

ssk2 committed:
(docs) add release note on changelog.md

ssk2 committed:
Update CHANGELOG.md.

rji commented on pom.xml:
```diff
@@ -0,0 +1,393 @@
+				<version>1.3.0</version>
+				<executions>
+					<execution>
+						<id>fetch-marathon-deploy</id>
```
This is in the Jenkins update center (version 1.3.1) so we can treat it as any of the other plugins now and not host it in our S3 bucket

rji commented on pom.xml:
```diff
@@ -0,0 +1,393 @@
+		<dependency>
+			<groupId>org.jenkins-ci.main</groupId>
+			<artifactId>jenkins-war</artifactId>
+			<version>1.651.2</version>
```
Can we bump this to 1.651.3?

rji commented on README.md:
```diff
@@ -15,33 +15,50 @@ up and running quickly with Jenkins on DCOS.
 ## Included in this repo
 Base packages:
-  * [Jenkins][jenkins-home] 1.642.4 (LTS)
+  * [Jenkins][jenkins-home] 1.651.2 (LTS)
```
Can we bump this to 1.651.3?

ssk2 committed:
Bump Jenkins version.

rji commented on scripts/release.sh:
```diff
@@ -0,0 +1,22 @@
+set -o errexit -o nounset -o pipefail
+
+# vars
+readonly DOCKER_IMAGE="mesosphere/velocity"
```
`s/velocity/jenkins/`

rji commented on scripts/release.sh:
```diff
@@ -0,0 +1,22 @@
+
+# generate sigs
+openssl md5 $TAR_FILE.gz >> velocity-$TAG.checksums
+openssl sha1 $TAR_FILE.gz >> velocity-$TAG.checksums
```
Same for these two lines: `s/velocity/jenkins/`

rji commented on scripts/release.sh:
```diff
@@ -0,0 +1,22 @@
+# vars
+readonly DOCKER_IMAGE="mesosphere/velocity"
+readonly TAG=${GIT_BRANCH##*/}
+readonly TAR_FILE="velocity-${TAG}.tar"
```
`s/velocity/jenkins/`

rji commented on .dockerignore:
```diff
@@ -3,4 +3,5 @@
 README.md
 demo/
 docs/
+scripts/build.sh
```
It looks like the script we want to ignore is actually `scripts/release.sh`.

ssk2 committed:
Pull Marathon plugin from JUC instead of wgeting it.

rji commented:
Before I forget, we should also take a look at any changes to the package on the private side and submit those to the public Universe (after we tag and push v1.0.0 here, of course)

colin-msphere commented on pom.xml:
```diff
@@ -0,0 +1,379 @@
+							<outputDirectory>${project.build.directory}/plugins</outputDirectory>
+						</artifactItem>
+						<artifactItem>
+							<groupId>org.jenkins-ci.plugins.workflow</groupId>
```
Should just be `org.jenkins-ci.plugins`

colin-msphere committed:
Fix marathon groupId

The marathon plugin had the wrong groupId and was unable to complete
the package operation.

Commit:
(maint) commit ccm test script to repo

Avoids storing test scripts in Jenkins.

colin-msphere commented on pom.xml:
```diff
@@ -0,0 +1,379 @@
+						</artifactItem>
+						<!-- Dependencies for Additional Plugins -->
+						<artifactItem>
+							<groupId>org.jenkins-ci.plugins</groupId>
```
```
[INFO] Configured Artifact: org.jenkins-ci.plugins:icon-shim:2.0.3:hpi
Downloading: http://repo.jenkins-ci.org/public/org/jenkins-ci/plugins/icon-shim/2.0.3/icon-shim-2.0.3.hpi
Downloading: http://repo.jenkins-ci.org/releases/org/jenkins-ci/plugins/icon-shim/2.0.3/icon-shim-2.0.3.hpi
Downloading: https://repo.maven.apache.org/maven2/org/jenkins-ci/plugins/icon-shim/2.0.3/icon-shim-2.0.3.hpi
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-dependency-plugin:2.10:copy (default) on project jenkins: Unable to find artifact. Could not find artifact org.jenkins-ci.plugins:icon-shim:hpi:2.0.3 in repo.jenkins-ci.org (http://repo.jenkins-ci.org/public/)
```

rji commented:
I'd like to do some manual testing on this tomorrow before merging, but otherwise this looks good to me.

ssk2 commented:
🚀
